pennylane_lightning/src/PyBind.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "pybind11/complex.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/numpy.h"
+#include "pybind11/stl.h"
+
+#include "Apply.hpp"
```

Comment:
Just wondering about having a separate `.hpp` file to this `.cpp` file instead of using `Apply.hpp` and having two/multiple `.cpp` files. Not super important would just adhere to convention, though this should work as is too.

Reply:
This file doesn't provide method bodies for anything in `Apply.hpp`, so its not so much sharing a `.hpp` file as it is using declarations in `Apply.hpp` (just like we'd include say `Util.hpp` if we wanted to use methods from that).

Personally, I don't consider it convention to have a `.hpp` file for source files that provide entry points (e.g. you wouldn't typically have a `main.hpp` if you had a `main.cpp`). I'm somewhat on the fence on whether we should even expose the `create()` and `apply()` methods defined here, or just make them `static` to make it clear they'll only be referenced in this source file.